brilirs/src/interp.rs

```diff
@@ -516,19 +622,58 @@ fn execute_value_op<T: std::io::Write>(
       let res = Value::Pointer(arg0.add(arg1));
       state.env.set(dest, res);
     }
+    Resolve => {
+      let res = match handlers.remove(&args[0]) {
```

Comment:
When I first saw this line, I thought it was a bug since you are the variable name(usize'd index) to find the handler which also happens to be a usize. I realize looking at the rest of this code that it probably isn't.

Still, this seems like a serious hit to modularity, each promise is tied to the variable name it is bound to so you can't call `id` on a promise, pass it as argument, or store it in an array. I think this should be documented. For what it's worth, I think this is a straight forward limitation to lift, though related to your scoping restriction somewhat.